src/main/java/com/rb/nonbiz/search/BinarySearchParameters.java

```diff
@@ -180,8 +180,9 @@ public void sanityCheckContents() {
       Y lowerValue = evaluatorOfX.apply(lowerBoundX);
       Y upperValue = evaluatorOfX.apply(upperBoundX);
       RBPreconditions.checkArgument(
-          comparatorForX.compare(lowerBoundX, upperBoundX) != 0,
-          "lower and upper bounds for X may be inverted; can also be ==, but that probably indicates a problem : %s and %s",
+          // can also be == ; that probably indicates a problem, but it's too conservative to throw an exception for.
```

Comment:
our application code would do well to throw an exception here, but since rbcommons is a general solution, it has to allow this corner case, because it's conceptually valid, even if it's possible this uncovers a problem.